mm2src/coins/eth.rs

```diff
@@ -4434,7 +4434,8 @@ impl EthCoin {
                 ))
             },
         };
-        let wallet_amount = u256_to_big_decimal(wallet_amount_uint, self.decimals)?;
+        // The "balanceOf" function in ERC1155 standard returns the exact count of tokens held by address without any decimals or scaling factors
+        let wallet_amount = wallet_amount_uint.to_string().parse::<BigDecimal>()?;
```

Comment:
In the PR comment you wrote
> ERC1155 "balanceOf" method started returning exact Nft amount owned by wallet without any scaling, so we dont need to use self.decimals shift when we convert U256 type to BigDecimal

So this wasn't the case before? I think it always returned exact count of tokens as `ERC1155` Tokens should be indivisible.

I don't think we should use `BigDecimal` here https://github.com/KomodoPlatform/komodo-defi-framework/blob/8b1170d9f33208e1c260d94d9ef9a7eb492e1047/mm2src/coins/nft/nft_structs.rs#L440-L441 So that decimals are rejected at the deserialization phase.

Reply:
> So this wasn't the case before?

Didnt have this issue generating tx with withdraw method, I used it myself to transfer erc1155 tokens between my own test addresses.

Reply:
updated amount type in withdraw_erc1155 rpc case
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2209/commits/a493f3f5321abff8e1dd39837441686f2e0fc015